nameko/events.py

```diff
@@ -183,10 +168,63 @@ def __init__(self, source_service, event_type, handler_type=SERVICE_POOL,
         super(EventHandler, self).__init__(
             queue=None, requeue_on_error=requeue_on_error)
 
+    @property
+    def broadcast_identifier(self):
+        """ A unique string to identify a service instance for `BROADCAST`
+        type handlers.
+
+        The `broadcast_identifier` is appended to the queue name when the
+        `BROADCAST` handler type is used. It must uniquely identify service
+        instances that receive broadcasts.
+
+        The default `broadcast_identifier` is a uuid that is set when the
+        service starts. It will change when the service restarts, meaning
+        that any unconsumed messages that were broadcast to the 'old' service
+        instance will not be received by the 'new' one. ::
+
+            @property
+            def broadcast_identifier(self):
+                # use a uuid as the identifier.
+                # the identifier will change when the service restarts and
+                # any unconsumed messages will be lost
+                return uuid.uuid4().hex
+
+        The default behaviour is therefore incompatible with reliable delivery.
+
+        An alternative `broadcast_identifier` that would survive service
+        restarts is ::
+
+            @property
+            def broadcast_identifier(self):
+                # use the machine hostname as the identifier.
+                # this assumes that only one instance of a service runs on
+                # any given machine
+                return socket.gethostname()
+
+        If neither of these approaches are appropriate, you could read the
+        value out of a configuration file ::
+
+            @property
+            def broadcast_identifier(self):
+                return self.config['SERVICE_IDENTIFIER']  # or similar
+
+        Broadcast queues are exclusive to ensure that `broadcast_identifier`
+        values are unique.
+        """
+        if self.reliable_delivery and self.handler_type is BROADCAST:
```

Comment:
I don't think it's strange to check both conditions are met before raising the error. Otherwise you're relying on it only being called in specific circumstances.

It doesn't matter if it's a method or a property. This would be weird:

```
>>> event_handler = EventHandler("svc", "evt")
>>> event_hander.get_broadcast_handler()
Traceback:
...
EventHandlerConfigurationError: You are using the default broadcast identifier
which is not compatible with reliable delivery...
```

It is less likely that you'd intentionally call `get_broadcast_handler` than you would `getmembers`, but I'm not sure that matters.